Allow user to plot correlation matrix for AECs

hwpang opened the pull request:
### Motivation or Problem
This is one of the changes for the BAC paper. Per Bill's request, we add a discussion regarding the correlation for both AEC and p-BAC.

### Description of Changes
I add a plotting function and allow the user to plot and save correlation matrix for AECs.

JacksonBurns commented on arkane/encorr/ae.py:
```diff
@@ -134,6 +144,49 @@ def execute(self, output_file: str = None):
+
+        ax.set_xticks(list(range(len(self.ae.correlation))))
+        ax.set_yticks(list(range(len(self.ae.correlation))))
+        ax.set_xticklabels(labels, fontsize=14)
```
make the font sizes passable as a dictionary, or don't set them (since they will be hard to make generally look correct)

hwpang replied:
Added!

JacksonBurns commented on arkane/encorr/ae.py:
```diff
@@ -125,6 +131,10 @@ def execute(self, output_file: str = None):
                     'atom_energies' if self.level_of_theory is None else self.level_of_theory))
 
+            if plot:
+                correlation_path = os.path.join(output_directory, f'AEC_{model_chemistry_formatted}_correlation.pdf')
```
add a unique timestamp to the end so you don't overwrite previous files by accident

hwpang replied:
I don't think this is necessary though because we put the level of theory name in the save path (`model_chemistry_formatted` ). Say if you fit for the same level of theory and plot it for multiple times because something wasn't quite right previously, I think as a user you would only care about the last one.

JacksonBurns replied:
RE in-person discussion: update arkane docs to state that previous files will be overwritten on subsequent runs

JacksonBurns commented on arkane/encorr/ae.py:
```diff
@@ -134,6 +144,49 @@ def execute(self, output_file: str = None):
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            return
```
please add `logging.error('cannot import matplotlib, original error'+str(e)`

JacksonBurns replied:
And the same in BACs

hwpang replied:
Added!

hwpang commented on arkane/encorr/ae.py:
```diff
@@ -134,6 +144,49 @@ def execute(self, output_file: str = None):
+            labels: Parameter labels.
+        """
+        try:
+            import matplotlib.pyplot as plt
```
Unbury this in both AEC and BAC

hwpang replied:
I look into this part of code again, and my new thought is that plotting functions are not necessary for the Arkane job execution to fail, so it makes sense to do this

JacksonBurns replied:
RE in-person discussion: make the import look like this: https://github.com/ReactionMechanismGenerator/RMG-Py/blob/main/rmgpy/ml/estimator.py#L121

hwpang commented:
> @hwpang can you add some documentation describing how to use this?

Thanks for the comment! Arkane's default behavior is to use all the plots, so this function is already being used in the default settings, unless the user passes in a flag indicating it to skip the plotting step (see [documentation](https://reactionmechanismgenerator.github.io/RMG-Py/users/arkane/running.html#suppressing-plot-creation)). What kind of documentation do you think these need? Happy to add a documentation but not sure what to add.

hwpang commented:
@JacksonBurns The new push rebased and addressed most of the comments! I also have some thoughts/questions for some of your comments (please see them above).

JacksonBurns reviewed (changes requested):
additional comments from in-person meeting

JacksonBurns reviewed (approved):
LGTM! Thanks for addressing the requested edits.
